### PR TITLE
Allow dot format choice

### PR DESCRIPTION
**Pull request by ssanderson**
Built on top of https://github.com/blaze/dask/pull/671.

### CODE REVIEW

File: dask/dot.py [hunk 1]
```diff
@@ -131,37 +132,82 @@ def to_graphviz(dsk, data_attributes=None, function_attributes=None):
...
+    arguments and returns None.
+    """
+    dummy = lambda *args, **kwargs: None
+    try:
+        import IPython.display as display
+    except ImportError:
+        # Can't return a display object if no IPython.
+        return dummy
+
+    if format in {'pdf', 'dot'}:
```
Comment (mrocklin) [inline dask/dot.py]: This is a py26 compatibility violation.  I suspect that using a tuple here would be just as good.

**Comment by mrocklin** [comment]
This generally looks quite good to me

**Comment by ssanderson** [comment]
Err... only the 3.4 build failed with

`dask/array/tests/test_linalg.py::test_svd_compressed /home/travis/build.sh: line 41:  2866 Segmentation fault      (core dumped) py.test dask --verbose`

Pretty sure this is unrelated to my change.  Intermittent failure?

**Comment by mrocklin** [comment]
Hrm, yes, unrelated.  I thought that I had resolved this issue in https://github.com/blaze/dask/pull/652 . It seems not.

**Comment by mrocklin** [comment]
There are some other intermittent failures lying about due to `dask.distributed` which will probably be evicted in the next few months.  My apologies for the dust.

**Comment by ssanderson** [comment]
Seems to have righted itself on the second attempt.

**Comment by mrocklin** [comment]
Grand.  Merging.

**Comment by shoyer** [comment]
Very nice, thanks!

**Pull request merged**